UnityMirror/UnityServer/Assets/Server/Scripts/AgentListener.cs

```diff
@@ -25,16 +25,23 @@ public class AgentListener : MonoBehaviour {
         UnityNetworkServer.Instance.OnPlayerRemoved.AddListener(OnPlayerRemoved);
 
         // get the port that the server will listen to
-        // we are getting the port via GSDK, the value of the "ListeningPortKey" must be the same value as the one you use when you create the Build
-        // or the one you use on LocalMultiplayerAgent JSON configuration file
         // We *have to* do it on process mode, since there might be more than one game server instances on the same VM and we want to avoid port collision
         // On container mode, we can omit the below code and set the port directly, since each game server instance will run on its own network namespace. However, below code will work as well
         // we have to do that on process
-        var config = PlayFabMultiplayerAgentAPI.GetConfigSettings();
-        if(config.ContainsKey(ListeningPortKey))
+        var connInfo = PlayFabMultiplayerAgentAPI.GetGameServerConnectionInfo();
+        // make sure the ListeningPortKey is the same as the one configured in your Build settings (either on LocalMultiplayerAgent or on MPS)
+        const string ListeningPortKey = "game_port";
+        var portInfo = connInfo.GamePortsConfiguration.Where(x=>x.Name == ListeningPortKey);
+        if(portInfo.Count() > 0)
         {
-            var port = int.Parse(config[ListeningPortKey]);
-            UnityNetworkServer.Instance.Port = port; // set the Mirror server to the port we got from GSDK
+            Debug.Log($"port_name was found in GSDK Config Settings.");
```

Comment:
Do we want to have in the debug message the value of the port number?

Reply:
+1